src/runtime/crt/host/microtvm_api_server.py

```diff
@@ -57,7 +71,14 @@ def server_info_query(self, tvm_version):
                     optional=["build"],
                     type="bool",
                     help="Run make with verbose output",
-                )
+                ),
+                server.ProjectOption(
+                    "memory_size_bytes",
+                    optional=["generate_project"],
+                    type="int",
+                    default=None,
```

Comment:
done

Reply:
If the option doesn't exist  _there is a default value_, because `options.get("memory_size_bytes", MEMORY_SIZE_BYTES)` is passed to `_populate_makefile` and so `MEMORY_SIZE_BYTES` will be used as the default value in the server, no?

The `default` metadatum is there not for the server, but mostly for the client, so if there is a default value that is used by the server if the option is not present, then it must be informed, as per:

https://github.com/apache/tvm-rfcs/blob/main/rfcs/0020-project_api_extend_metadata.md#on-default-metadatum

Reply:
@mehrdadh Maybe we can improve this situation by processing the options based on `PROJECT_OPTIONS` and taking `default` field into account.

Reply:
@gromero yeah that makes sense